### PR TITLE
Add LARC implementation to Caffe2

### PR DESCRIPTION
**Pull request by lilinyy09**
Summary: This diff added layer-wise adapted learning rate control (LARC) to Caffe2. LARC optimization enables us to train larger batch size within the same number of epochs without significant loss in accuracy.

Differential Revision: D8959161

### CODE REVIEW

**Comment by ezyang** [comment]
abandoned in fbcode
